### ISSUE TITLE
Problems with before_log?

### ISSUE THREAD

**Issue by screamff** [jd/tenacity#264]
python version:3.7.2 32bit
platform:win 10
tenacity version:6.2.0

I found before_log code as below:
```
def before_log(logger, log_level):
    """Before call strategy that logs to some logger the attempt."""
    def log_it(retry_state):
        logger.log(log_level,
                   "Starting call to '%s', this is the %s time calling it.",
                   _utils.get_callback_name(retry_state.fn),
                   _utils.to_ordinal(retry_state.attempt_number))

    return log_it
```
Then i used logger in loguru, pass `before=before_log(logger.log, 'DEBUG')` to retry, the result:
```
2020-11-03 15:24:56.091 | DEBUG    | __main__:log_it:11 - Starting call to '%s', this is the %s time calling it.
```

Then I changed the code:
```
def before_log(logger, log_level):
    """Before call strategy that logs to some logger the attempt."""
    def log_it(retry_state):
        logger.log(log_level,
                   "Starting call to '%s', this is the %s time calling it." %(
                   _utils.get_callback_name(retry_state.fn),
                   _utils.to_ordinal(retry_state.attempt_number)))

    return log_it
```

It succeed:
```
2020-11-03 15:32:34.746 | DEBUG    | tenacity.before:log_it:30 - Starting call to '__main__.raise_my_exception', this is the 1st time calling it.
```

**Comment by screamff** [comment]
I used standard module logging and it works alright with whether raw `before_log` method or my modified code

**Comment by asqui** [comment]
I've not heard of `loguru` before but this seems to be because it uses [Modern string formatting using braces style
](https://github.com/Delgan/loguru#modern-string-formatting-using-braces-style) which is listed as one of the features, but unfortunately makes it incompatible with the standard library logger.

Compare [`loguru._logger.Logger.log()`](https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger.log) (which makes reference to `string.format()`) with [`logging.Logger.log()`](https://docs.python.org/2/library/logging.html#logging.Logger.log) (which makes reference to "the string formatting operator").

**Comment by screamff** [comment]
So if I want to use loguru, I just need to monkeypatch `before_log` and `after_log`?Is there any other side effect by doing this? 
 @asqui

**Comment by asqui** [comment]
Looks like this issue would be addressed by PR #294, if it gets merged.

**Comment by asqui** [comment]
@jd I think this is resolved now that #294 got merged?

**Comment by jd** [comment]
Yes!
